### PR TITLE
Chore/move Gradle from Groovy to Kotlin

### PR DESCRIPTION
**Pull request by IdanKoblik**

### CODE REVIEW

File: build-logic/build.gradle.kts [hunk 1]
```diff
@@ -0,0 +1,15 @@
+plugins {
+    kotlin("jvm") version "1.9.20"
+    `kotlin-dsl`
+}
+
+group = "net.apartium.cocoa-beans"
```
Comment (liorsl) [inline build-logic/build.gradle.kts:6]: inherit from parent

File: build-logic/build.gradle.kts [hunk 2]
```diff
@@ -0,0 +1,15 @@
...
+version = "1.0-SNAPSHOT"
```
Comment (liorsl) [inline build-logic/build.gradle.kts:7]: inherit from parent

File: build-logic/settings.gradle.kts [hunk 3]
```diff
@@ -0,0 +1 @@
+rootProject.name = "build-logic"
```
Comment (liorsl) [inline build-logic/settings.gradle.kts:1]: delete this file

File: build.gradle.kts [hunk 4]
```diff
@@ -0,0 +1,94 @@
...
+    id("com.github.johnrengelman.shadow") version "8.1.1"
+    id("io.papermc.hangar-publish-plugin") version "0.1.2"
+}
+
+
+val snapshot = System.getProperty("apartium.snapshot", "true").toBoolean()
+val isCi = System.getenv("GITHUB_ACTOR") != null
+
+group = "net.apartium.cocoa-beans"
+version = "0.0.17" + (if (snapshot) "-SNAPSHOT" else "")
```
Comment (liorsl) [inline build.gradle.kts:13]: take latest version from main

File: build-logic/build.gradle.kts [hunk 5]
```diff
@@ -0,0 +1,15 @@
...
+
+repositories {
```
Comment (liorsl) [inline build-logic/build.gradle.kts:9]: delete this block

File: build.gradle.kts [hunk 6]
```diff
@@ -0,0 +1,94 @@
...
+
+    tasks {
+        test {
+            useJUnitPlatform()
+        }
+    }
+
+}
+
+hangarPublish {
```
Comment (liorsl) [inline build.gradle.kts:69]: check if this works

**Comment by ikfir** [comment]
@IdanKoblik `gradle properties`

**Pull request merged**